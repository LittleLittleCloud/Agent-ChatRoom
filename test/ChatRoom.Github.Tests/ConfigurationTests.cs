using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using ChatRoom.OpenAI;
using FluentAssertions;
using Json.Schema;
using Xunit;
using Json.Schema.Generation;

namespace ChatRoom.Github.Tests;

public class ConfigurationTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void VerifyConfigurationSchema()
    {
        var schema = new JsonSchemaBuilder()
            .FromType<GithubConfiguration>()
            .Build();
        var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });
        Approvals.Verify(json);

        var schemaFileName = "chatroom_github_configuration_schema.json";
        var schemaFilePath = Path.Join("Schema", schemaFileName);
        var schemaFile = File.ReadAllText(schemaFilePath);
        schemaFile.Should().BeEquivalentTo(json);
    }
}
