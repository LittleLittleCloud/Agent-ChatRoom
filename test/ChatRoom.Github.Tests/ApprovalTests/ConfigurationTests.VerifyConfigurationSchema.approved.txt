{
  "type": "object",
  "properties": {
    "IssueHelper": {
      "type": "object",
      "properties": {
        "name": {
          "$ref": "#/$defs/string"
        },
        "system_message": {
          "$ref": "#/$defs/string1"
        },
        "description": {
          "$ref": "#/$defs/string2"
        }
      },
      "description": "Issue helper configuration"
    },
    "github_token": {
      "type": "string",
      "description": "GitHub token, will use $env:GITHUB_TOKEN if not provided"
    },
    "openai_config": {
      "type": "object",
      "properties": {
        "llm_type": {
          "$ref": "#/$defs/lLMType"
        },
        "openai_api_key": {
          "$ref": "#/$defs/string3"
        },
        "openai_model_id": {
          "$ref": "#/$defs/string4"
        },
        "azure_openai_endpoint": {
          "$ref": "#/$defs/string5"
        },
        "azure_openai_key": {
          "$ref": "#/$defs/string6"
        },
        "azure_openai_deploy_name": {
          "$ref": "#/$defs/string7"
        },
        "third_party_llm_endpoint": {
          "$ref": "#/$defs/string8"
        },
        "third_party_llm_key": {
          "$ref": "#/$defs/string9"
        },
        "third_party_llm_model_id": {
          "$ref": "#/$defs/string10"
        },
        "ModelId": {
          "$ref": "#/$defs/string11"
        }
      },
      "description": "OpenAI configuration"
    }
  },
  "$defs": {
    "string": {
      "type": "string",
      "description": "Name of the issue helper agent, default is \u0027issue-helper\u0027"
    },
    "string1": {
      "type": "string",
      "description": "System message, default is \u0027You are a github issue helper\u0027"
    },
    "string2": {
      "type": "string",
      "description": "Agent description, default is \u0027I am a github issue helper, I can help you with your github issues.\u0027"
    },
    "lLMType": {
      "enum": [
        "AOAI",
        "OpenAI",
        "ThirdParty"
      ],
      "description": "LLM provider, default is Azure OpenAI"
    },
    "string3": {
      "type": "string",
      "description": "OpenAI API key, default is $env:OPENAI_API_KEY"
    },
    "string4": {
      "type": "string",
      "description": "OpenAI model ID, default is gpt-3.5-turbo"
    },
    "string5": {
      "type": "string",
      "description": "Azure OpenAI endpoint, default is $env:AZURE_OPENAI_ENDPOINT"
    },
    "string6": {
      "type": "string",
      "description": "Azure OpenAI key, default is $env:AZURE_OPENAI_API_KEY"
    },
    "string7": {
      "type": "string",
      "description": "Azure OpenAI deploy name, default is $env:AZURE_OPENAI_DEPLOY_NAME"
    },
    "string8": {
      "type": "string",
      "description": "Third-party LLM provider endpoint, default is $env:THIRD_PARTY_LLM_ENDPOINT"
    },
    "string9": {
      "type": "string",
      "description": "Third-party LLM provider key, default is $env:THIRD_PARTY_LLM_KEY"
    },
    "string10": {
      "type": "string",
      "description": "Third-party LLM provider model ID, default is $env:THIRD_PARTY_LLM_MODEL_ID"
    },
    "string11": {
      "type": "string",
      "readOnly": true
    }
  }
}